### PR TITLE
feat: update ibc-go to v4, adapt packages and API

### LINKED ISSUE #8223: Upgrade to ibc-go v4
Entails upgrading:

- tendermint to cometbft 0.34.27
- cosmos-sdk to 0.45.16
- ibc-go to 4.4.2
-

### PR DESCRIPTION
**Pull request by JimLarson**
closes: #8223

## Description

Upgrade to ibc-go v4.4.2. Entails upgrades of cosmos-sdk to v0.45.16 and tendermint to cometbft v0.34.27.

### Security Considerations

Closes any security issues that these upgrades address.

### Scaling Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

See test plan document.

### Upgrade Considerations

Yes.

### CODE REVIEW

File: golang/cosmos/cmd/runtillmsg/main.go
Comment (michaelfig) [inline golang/cosmos/cmd/runtillmsg/main.go:1]: Cool! So much easier.

**Review by michaelfig — APPROVED** [review]

**Comment by mhofman** [comment]
Can we merge https://github.com/agoric-labs/cometbft/pull/5 first and only update the cometbft dependency once.

File: golang/cosmos/go.mod [hunk 1]
```diff
@@ -142,14 +155,11 @@ replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
 // At least until post-v0.34.14 is released with
 // https://github.com/tendermint/tendermint/issue/6899 resolved.
-replace github.com/tendermint/tendermint => github.com/agoric-labs/tendermint v0.34.23-alpha.agoric.3
+replace github.com/tendermint/tendermint => github.com/agoric-labs/cometbft v0.34.27-alpha.agoric.2
```
Comment (mhofman) [inline golang/cosmos/go.mod]: New version has been tagged with the `BlockParams.MaxBytes` change

```suggestion
replace github.com/tendermint/tendermint => github.com/agoric-labs/cometbft v0.34.27-alpha.agoric.3
```

Comment (JimLarson) [reply golang/cosmos/go.mod]: Done.

**Comment by JimLarson** [comment]
PTAL @michaelfig @mhofman

**Review by michaelfig — APPROVED** [review]

**Pull request merged**